test/check-application

```diff
@@ -944,6 +948,255 @@ class TestFiles(testlib.MachineCase):
         b.wait_visible("[data-item='newfile2']")
         b.click(".breadcrumb-button:nth-of-type(3)")
 
+    @testlib.skipBrowser(".upload_files() doesn't work on Firefox", "firefox")
+    @testlib.skipImage("Debian-testing has Cockpit 311 without new upload support in fsreplace1", "debian-testing")
+    def testUpload(self):
+        b = self.browser
+        m = self.machine
+
+        self.enter_files()
+
+        def get_piechart_progress(sel):
+            b.wait_visible(sel)
+            style = b.attr(sel, "style")
+            m = re.search(r"--progress: (\d+).\d+%;", style)
+            assert m is not None
+            return int(m.group(1))
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # Test cancelling of download
+            big_file = str(Path(tmpdir) / "bigfile.img")
+            subprocess.check_call(["truncate", "-s", "1500MB", big_file])
+
+            b.click("#upload-file-btn")
+            b.upload_files("#upload-file-btn + input[type='file']", [big_file])
+            b.wait_visible("#upload-file-btn:disabled")
+
+            # Wait for some progress and cancel
+            b.click("#upload-progress-btn")
+            b.wait(lambda: get_piechart_progress("#upload-progress-btn") >= 2)
+            b.wait_in_text(".upload-progress-0", "bigfile.img")
+            b.wait(lambda: b.get_pf_progress_value(".upload-progress-0") >= 2)
+
+            b.click(".cancel-button-0")
+
+            b.wait_not_present("#upload-progress-btn")
+            b.wait_visible("#upload-file-btn")
+            b.wait_in_text(".pf-v5-c-alert__description", "Cancelled upload of bigfile.img")
+            b.click(".pf-v5-c-alert__action button")
+            b.wait_not_present(".pf-v5-c-alert__action")
```

Comment:
This could check that bigfile.img got removed on the server after cancelling.

Reply:
As fsreplace1 makes a random file, let's check for the amount of files, it should not change.